src/Composer/Command/BumpCommand.php

```diff
@@ -184,7 +196,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $lock->write($lockData);
         }
 
-        return 0;
+        return $dryRun && $changeCount > 0 ? self::ERROR_GENERIC : 0;
     }
```

Comment:
I'd probably prefer to avoid ternary here:

```suggestion
        if ($dryRun && $changeCount > 0) {
            return self::ERROR_GENERIC;
        }

        return 0;
    }
```